src/Wrapper/GkmsClientFactory.php

```diff
@@ -4,27 +4,62 @@
 
 namespace Keboola\ObjectEncryptor\Wrapper;
 
+use Google\Auth\HttpHandler\HttpHandlerFactory;
 use Google\Cloud\Kms\V1\KeyManagementServiceClient;
+use GuzzleHttp\BodySummarizer;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
 use Keboola\ObjectEncryptor\EncryptorOptions;
 use Keboola\ObjectEncryptor\Exception\ApplicationException;
 use Throwable;
 
 class GkmsClientFactory
 {
+    public const TIMEOUT = 120;
+    public const CONNECT_TIMEOUT = 10;
+
     public function createClient(EncryptorOptions $encryptorOptions): KeyManagementServiceClient
     {
         /* It seems that KeyManagementServiceClient client does not accept retrySettings configuration
             (as some other GCP clients do), therefore we disable retries completely and rely on application level
             retries in GenericGKMWrapper. */
         try {
+            // Create the handler with passed http client with timeout settings
+            $handler = HttpHandlerFactory::build($this->createHttpClient());
+
             // GKM client checks for authorization when created, authorization is cached in memory
             return new KeyManagementServiceClient(
                 [
                     'disableRetries' => true,
+                    'transportConfig' => [
+                        'rest' => [
+                            'httpHandler' => [$handler, 'async'],
+                        ],
+                    ],
                 ]
             );
         } catch (Throwable $e) {
             throw new ApplicationException('Cipher key settings are invalid: ' . $e->getMessage(), 0, $e);
         }
     }
+
+    private function createHttpClient(): Client
+    {
+        // copied from HttpHandlerFactory::build()
```

Comment:
no kdybys tajk dal ještě git link, protože tohle je obskuritka (i když efekt je asi plusmínus zřejmej)

Reply:
Jo, byl jsem linej :)

https://github.com/googleapis/google-auth-library-php/blob/3d68b6d5161900abb873f252f4106f9a6aa2a230/src/HttpHandler/HttpHandlerFactory.php#L34

Reply:
Jinak tohleto upraveni (zvetseni) delky tela pri errorech by se nam mozna taky hodilo nekde v nasich klientech. 
Ze by to neurezavalo.

Reply:
Ten link by se hodil v kodu u toho commentu :wink: 

V SAPI clientovi jsme tu delku uz presne tkale resili https://github.com/keboola/storage-api-php-client/blob/c28f6702028a635a16c4c5b6b8052331b0131f24/src/Keboola/StorageApi/HandlerStack.php#L31